tests/test_utils/tests.py

```diff
@@ -872,6 +873,40 @@ class MyCustomField(IntegerField):
         self.assertFieldOutput(MyCustomField, {}, {}, empty_value=None)
 
 
+class AssertSignalSentTests(TestCase):
+    def test_signal_sent(self):
+        with self.assertSignalSent(signals.post_save):
+            Car.objects.create(name='Were-Car')
+
+    def test_signal_sent_arguments(self):
+        with self.assertSignalSent(signals.post_save, required_kwargs=['sender']):
+            Car.objects.create(name='Original Were-Car')
+
+    def test_signal_sent_inspect(self):
+        with self.assertSignalSent(signals.post_save) as cm:
+            Car.objects.create(name='Not a Were-Car')
+            self.assertEqual(
+                cm.received_kwargs['sender'].__class__,
+                Car.__class__
+            )
+
+    @unittest.expectedFailure
+    def test_signal_sent_fail(self):
+        with self.assertSignalSent(signals.post_save) as cm:
+            Car(name='A Were-Car that cannot be saved')
+            self.assertEqual(cm.received_kwargs, {})
+            self.assertFalse(cm.signal_sent)
+
+    def test_signal_not_sent(self):
+        with self.assertSignalNotSent(signals.post_save):
+            Car(name='Another Were-Car that cannot be saved.')
+
+    @unittest.expectedFailure
```

Comment:
We usually try to keep `expectedFailure` usages for tests we plan to fix in the future as the test runner doesn't consider _unexpected success_ as failures. Using `self.assertRaisesMessage(AssertionError, 'Signal was unexpectedly sent.')` would be more appropriate here.